usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopAll.java

```diff
@@ -0,0 +1,304 @@
+ */
+public class ProcessParsedStopAll {
+
+    /**
+     * This enum matches types that can be found in the location_type field of stops.txt
+     * // see https://gtfs.org/reference/static#stopstxt
+     * It's used to decide which concrete type derived from {@link LocationBase} to instantiate
+     */
+    private enum LocationType {
+        STOP_OR_PLATFORM(0),
+        STATION(1),
+        ENTRANCE(2),
+        GENERIC_NODE(3),
+        BOARDING_AREA(4);
+
+        private int value;
+
+        LocationType(int value) {
+            this.value = value;
+        }
+
+        static public LocationType fromInt(Integer locationTypeAsInt) {
+            if (locationTypeAsInt == null) {
+                return STOP_OR_PLATFORM;
+            }
+            return Stream.of(LocationType.values())
+                    .filter(enumItem -> enumItem.value == locationTypeAsInt)
+                    .findAny()
+                    .orElse(STOP_OR_PLATFORM);
+        }
+    }
+
+    private final ValidationResultRepository resultRepository;
+    private final GtfsDataRepository gtfsRepo;
+    private final StopOrPlatform.StopOrPlatformBuilder stopOrPlatformBuilder;
+    private final Station.StationBuilder stationBuilder;
+    private final Entrance.EntranceBuilder entranceBuilder;
+    private final GenericNode.GenericNodeBuilder genericNodeBuilder;
+    private final BoardingArea.BoardingAreaBuilder boardingAreaBuilder;
+
+    public ProcessParsedStopAll(final ValidationResultRepository resultRepo,
+                                final GtfsDataRepository gtfsRepo,
+                                final StopOrPlatform.StopOrPlatformBuilder stopOrPlatformBuilder,
+                                final Station.StationBuilder stationBuilder,
+                                final Entrance.EntranceBuilder entranceBuilder,
+                                final GenericNode.GenericNodeBuilder genericNodeBuilder,
+                                final BoardingArea.BoardingAreaBuilder boardingAreaBuilder) {
+        this.resultRepository = resultRepo;
+        this.gtfsRepo = gtfsRepo;
+        this.stopOrPlatformBuilder = stopOrPlatformBuilder;
+        this.stationBuilder = stationBuilder;
+        this.entranceBuilder = entranceBuilder;
+        this.genericNodeBuilder = genericNodeBuilder;
+        this.boardingAreaBuilder = boardingAreaBuilder;
+    }
+
+    public void execute(final Map<String, ParsedEntity> parsedEntityByStopId) {
+        Map<String, List<String>> childrenPerStationId = new HashMap<>();
+
+        parsedEntityByStopId.values().forEach(stop -> {
+            String childId = stop.getEntityId();
+            String parentId = (String) stop.get("parent_station");
+            if (parentId != null) {
+                if (childrenPerStationId.containsKey(parentId)) {
+                    childrenPerStationId.get(parentId).add(childId);
+                } else {
+                    List<String> childrenList = new ArrayList<>();
```

Comment:
Should we work with `HashSet` to ensure no duplicate?

Reply:
I think we check for duplicate stop_ids elsewhere when inserting into the repo? If so, then not for duplicate checks (because a duplicate record would be the only way it could end up with the same child twice).

But HashSet would be useful here for efficient lookup to see if a stop has a particular stop_id as a child - although I don't know if we would encounter that use case in practice.

Reply:
It seems that would happen only in the case of duplicated rows in the original dataset (same `stop_id`, same `parent_station`) and would be caught as such.